test/webserver.mjs

```diff
@@ -82,27 +84,10 @@ class WebServer {
   }
 
   async #handler(request, response) {
-    // Validate and parse the request URL.
-    const url = request.url.replaceAll("//", "/");
-    const urlParts = /([^?]*)((?:\?(.*))?)/.exec(url);
-    let pathPart;
-    try {
-      // Guard against directory traversal attacks such as
-      // `/../../../../../../../etc/passwd`, which let you make GET requests
-      // for files outside of `this.root`.
-      pathPart = path.normalize(decodeURI(urlParts[1]));
-      // `path.normalize` returns a path on the basis of the current platform.
-      // Windows paths cause issues in `checkRequest` and underlying methods.
-      // Converting to a Unix path avoids platform checks in said functions.
-      pathPart = pathPart.replaceAll("\\", "/");
-    } catch {
-      // If the URI cannot be decoded, a `URIError` is thrown. This happens for
-      // malformed URIs such as `http://localhost:8888/%s%s` and should be
-      // handled as a bad request.
-      response.writeHead(400);
-      response.end("Bad request", "utf8");
-      return;
-    }
+    // URLs are normalized and automatically disallow directory traversal
+    // attacks. For example, http://HOST:PORT/../../../../../../../etc/passwd
+    // is equivalent to http://HOST:PORT/etc/passwd.
+    const url = new URL(`http://${this.host}:${this.port}${request.url}`);
```

Comment:
By removing the old "hacks" used here, this *may* also address https://bugzilla.mozilla.org/show_bug.cgi?id=1904262